test/ffmpeg-qsv/encode/avc.py

```diff
@@ -220,3 +220,31 @@ def test(self, case, bframes, bitrate, fps, quality, refs, profile, ladepth):
       ladepth   = ladepth,
     )
     self.encode()
+
+class forced_idr(AVCEncoderTest):
+  def init(self, tspec, case, rcmode, bitrate, qp, quality, profile):
+    vars(self).update(tspec[case].copy())
+    if "cqp" == rcmode:
+        bitrate = None
+        maxrate = None
```

Comment:
User config should not set bitrate or maxrate in this rcmode.  The parameter gen function should set the default to None if not defined in user config, otherwise pass the user config setting to the test function.  Then we can just assert here if they are still not None.

Similar should apply to other rcmode where parameter does not make sense.